aspnetcore/aspnetcore-2.1.md

```diff
@@ -0,0 +1,221 @@
+            // Assert
+            Assert.Equal(HttpStatusCodes.OK, response.StatusCode);
+        }
+    }
+}
+```
+
+For more information, see the [Announcement on GitHub](https://github.com/aspnet/announcements/issues/275)
+
+## [ApiController], ActionResult
+
+In ASP.NET Core 2.1, improved support for OpenAPI specification(previously known as **Swaggger**)) has been added to Web API. `ActionResult<T>` is a new type added to allow an application to return either a response type or any other action result (similar to IActionResult), while still indicating the response type. The `[ApiController] attribute has also been added as the way to opt-in to Web API specific conventions and behaviors.
+
+For more information, see [Build wbe APIs with ASP.NET Core](xref:web-api)
+
+## IHttpClientFactory
+
+The new `HttpClientFactory` type can be registered and used to configure and consume instances of HttpClient in applications. HttpClient already has the concept of delegating handlers that could be linked together for outgoing HTTP requests. The factory will make registering of these per named client more intuitive as well as implement a Polly handler that allows Polly policies to be used for Retry, CircuitBreakers, etc.
+
+For more information, see [Initiate HTTP Requests](xref:fundamentals/http-requests)
+
+## Kestrel on Sockets
+
+In ASP.NET Core 2.1, a new socket transport has been added to Kestrel. An application can be configured to use this new transport by adding the `.UseSockets()` extension to the `HostBuilder`
+
+```csharp
+public static IWebHost BuildWebHost(string[] args) =>
+    WebHost.CreateDefaultBuilder(args)
+    .UseSockets()
+    .UseStartup<Startup>()
+ ```
+
+## Generic host builder
+
+In ASP.NET Core 2.1, the Generic Host Builder (`HostBuilder`) has been introduced. This Builder can be used for applications that do not process HTTP requests (Messaging, background tasks, etc).
+
+For more information, see [.NET Generic Host](xref:fundamentals/host/generic)
+
+## Updated SPA templates
+
+In ASP.NET Core 2.1, the Single Page Application Templates for Angular, React, and React with Redux have been updated to account for changes in the architecture of each framework. 
+For more information, see [Use the Single Page Application templates with ASP.NET Core](xref:spa)
+
+## Microsoft.AspNetCore.App package
+
+ASP.NET Core 2.1 will introduce a new meta-package for use by applications: `Microsoft.AspNetCore.App`. The new meta-package differs from the existing meta-package in that it reduces the number of dependencies of packages not owned or supported by the ASP.NET or .NET teams to just those deemed necessary to ensure the major framework features function. The `existing Microsoft.AspNetCore.All` meta-package will continue to be made available throughout the 2.x lifecycle. For additional details see https://github.com/aspnet/Announcements/issues/287.
+
+## Other documentation updates for 2.0
```

Comment:
I don't think this section makes sense. This doc is for changes in the framework, not for changes in the docs.